File: app/src/main/java/ru/dombuketa/filmslocaror/utils/MessageReceiver.java
package ru.dombuketa.filmslocaror.utils;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.widget.Toast;

public class MessageReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        if (intent == null) return;
        switch (intent.getAction()){
            case Intent.ACTION_BATTERY_LOW:
                Toast.makeText(context,"Батарея разряжена", Toast.LENGTH_SHORT).show();
                break;
            case Intent.ACTION_POWER_CONNECTED:
                Toast.makeText(context,"Батарея заряжается", Toast.LENGTH_SHORT).show();
        }

    }
}

File: app/src/main/java/ru/dombuketa/filmslocaror/utils/MessageReceiver.kt
package ru.dombuketa.filmslocaror.utils

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.widget.Toast
import androidx.appcompat.app.AppCompatDelegate

class MessageReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context?, intent: Intent?) {
        if (intent == null) return
        when (intent.action) {
            Intent.ACTION_BATTERY_LOW -> {
                Toast.makeText(context, "Батарея разряжена", Toast.LENGTH_SHORT).show()
                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES)
            }
            Intent.ACTION_POWER_CONNECTED -> {
                Toast.makeText(context, "Батарея заряжается", Toast.LENGTH_SHORT).show()
                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO)
            }
            Intent.ACTION_AIRPLANE_MODE_CHANGED -> if (AppCompatDelegate.getDefaultNightMode() != AppCompatDelegate.MODE_NIGHT_YES) {
                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES)
                Toast.makeText(context, "Ночной режим", Toast.LENGTH_SHORT).show()
            } else {
                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO)
                Toast.makeText(context, "Дневной режим", Toast.LENGTH_SHORT).show()
            }
        }
    }
}